build.wake

```diff
@@ -36,9 +36,41 @@ export def install: List String => Result String Error = match _
     dest, Nil = doInstall (in cwd dest) "default" | rmap (\_ "INSTALL")
     _ = Fail "no directory specified (try: install /opt/local)".makeError
 
+def bootstrapTarget targetFn variant =
```

Comment:
I have no idea how you'll feel about this but it seemed like a neat solution @JakeSiFive 

I actually had the thought that it would be cool to gut the Makefile as much as possible. Maybe we could have the Makefile just build wake and shim-wake, then run a bootstrap build flow where everything is build it wake (would require `setLocalPlanOnly True` since we wouldn't have wakebox yet)